docs/threading.rst

```diff
@@ -1,33 +1,42 @@
 Threading model
 ===============
 
-The *Connector* API does not implement any mechanism for
-thread safety. Applications are responsible for protecting concurrent calls to
-the Connector API. In Python, you will have to protect the calls to
-*Connector* if you are using different threads.
+.. py:currentmodule:: rticonnextdds_connector
 
-The following section shows an example.
+.. testsetup:: *
 
-TODO: review, complete section
+   import rticonnextdds_connector as rti
+
+RTI Connector only allows the concurrent use of different :class:`Connector`
+instances. Method calls to the same :class:`Connector` instance or any of its contained
+entities are not thread safe, and the application is responsible to protect them
+if they're used concurrently.
 
 Protecting calls to *Connector*
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 The following example shows how to use the Python ``threading`` package to
-protect calls to *Connector*, one of multiple ways to do so::
+protect calls to the same :class:`Connector`:
+
+.. testcode::
 
    import threading
 
-   sem = threading.Semaphore()
+   connector = rti.Connector("MyParticipantLibrary::MyParticipant", "ShapeExample.xml")
+   lock = threading.RLock()
```

Comment:
I have never tested this (and I don't think we have any unit tests covering it) - have you?

Reply:
I've tested it by hand and with `make test`. We don't need a unit test since this isn't RTI code